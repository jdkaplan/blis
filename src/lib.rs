pub mod bytecode;
pub mod compile;
pub mod parse;
pub mod runtime;
pub mod vm;

#[cfg(test)]
mod testing;
