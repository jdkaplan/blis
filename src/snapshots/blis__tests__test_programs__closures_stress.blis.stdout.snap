---
source: src/lib.rs
description: "func main() {\n    func outer(x) {\n        println(\"outer\");\n\n        func middle(name) {\n            println(\"middle\", name);\n\n            func inner() {\n                println(\"inner\", name);\n                println(x);\n                x = x + 1;\n                println(x);\n                println(\"-----\");\n            }\n\n            inner\n        }\n\n        middle\n    }\n\n    let middle = outer(0);\n    let inner1 = middle(\"first\");\n    inner1();\n    inner1();\n\n    let inner2 = middle(\"second\");\n    inner2();\n\n    middle = nil;\n\n    println(inner1);\n    inner1();\n    inner1 = nil;\n\n    println(inner2);\n    inner2();\n    inner2 = nil;\n}\n\nmain();\n"
input_file: test_programs/closures_stress.blis
---
outer
middle first
inner first
0
1
-----
inner first
1
2
-----
middle second
inner second
2
3
-----
<closure inner 0x000000000001>
inner first
3
4
-----
<closure inner 0x000000000002>
inner second
4
5
-----
